src/D2L.Bmx/Okta/OktaClient.cs

```diff
@@ -1,62 +1,72 @@
 using System.Net;
-using System.Net.Http.Headers;
 using System.Net.Http.Json;
 using System.Text.Json;
 using D2L.Bmx.Okta.Models;
 
 namespace D2L.Bmx.Okta;
 
-internal interface IOktaApi {
-	void SetOrganization( string organization );
-	void AddSession( string sessionId );
+internal interface IOktaClientFactory {
+	IOktaAnonymousClient CreateAnonymousClient( string org );
+	IOktaAuthenticatedClient CreateAuthenticatedClient( string org, string sessionId );
+}
+
+internal interface IOktaAnonymousClient {
 	Task<AuthenticateResponse> AuthenticateAsync( string username, string password );
 	Task IssueMfaChallengeAsync( string stateToken, string factorId );
 	Task<AuthenticateResponse> VerifyMfaChallengeResponseAsync(
 		string stateToken,
 		string factorId,
-		string challengeResponse );
+		string challengeResponse
+	);
 	Task<OktaSession> CreateSessionAsync( string sessionToken );
-	Task<OktaApp[]> GetAwsAccountAppsAsync();
-	Task<string> GetPageAsync( string samlLoginUrl );
 }
 
-internal class OktaApi : IOktaApi {
-	private readonly CookieContainer _cookieContainer;
-	private readonly HttpClient _httpClient;
-	private string? _organization;
+internal interface IOktaAuthenticatedClient {
+	Task<OktaApp[]> GetAwsAccountAppsAsync();
+	Task<string> GetPageAsync( string url );
+}
 
-	public OktaApi() {
-		_cookieContainer = new CookieContainer();
-		_httpClient = new HttpClient( new HttpClientHandler { CookieContainer = _cookieContainer } ) {
-			Timeout = TimeSpan.FromSeconds( 30 )
+internal class OktaClientFactory : IOktaClientFactory {
+	IOktaAnonymousClient IOktaClientFactory.CreateAnonymousClient( string org ) {
+		var httpClient = new HttpClient {
+			Timeout = TimeSpan.FromSeconds( 30 ),
+			BaseAddress = GetBaseAddress( org ),
 		};
-		_httpClient.DefaultRequestHeaders.Accept.Add( new MediaTypeWithQualityHeaderValue( "application/json" ) );
```

Comment:
I didn't actually mean to remove it, but now that I noticed, I'm not sure I'd add it back:
* Okta API works fine without this header.
* This header is a lie when used in the [`GetPageAsync`](https://github.com/Brightspace/bmx/pull/477/files#diff-68472d07d2f2a8650bacbb76d8a2511a58d8d662adf585119f1299282196441cL181-R191) method (we expect HTML there)